parachain/pallets/ethereum-beacon-light-client/src/lib.rs

```diff
@@ -0,0 +1,710 @@
+
+			Self::verify_header(
+				block_root,
+				update.finality_branch,
+				update.attested_header.state_root,
+				FINALIZED_ROOT_DEPTH,
+				FINALIZED_ROOT_INDEX,
+			)?;
+
+			let current_period = Self::compute_current_sync_period(update.attested_header.slot);
+
+			let sync_committee = <SyncCommittees<T>>::get(current_period);
+
+			if (SyncCommittee { pubkeys: vec![], aggregate_pubkey: PublicKey([0; 48]) }) == sync_committee {
+				return Err(Error::<T>::SyncCommitteeMissing.into());
+			}
+
+			let genesis = <ChainGenesis<T>>::get();
+
+			Self::verify_signed_header(
+				sync_committee_bits,
+				update.sync_aggregate.sync_committee_signature,
+				sync_committee.pubkeys,
+				update.fork_version,
+				update.attested_header,
+				genesis.validators_root,
+			)?;
+
+			Self::store_header(block_root, update.finalized_header);
+
+			Ok(())
+		}
+
+		pub(super) fn verify_signed_header(
+			sync_committee_bits: Vec<u8>,
+			sync_committee_signature: Vec<u8>,
+			sync_committee_pubkeys: Vec<PublicKey>,
+			fork_version: ForkVersion,
+			header: BeaconBlockHeader,
+			validators_root: H256,
+		) -> DispatchResult {
+			let mut participant_pubkeys: Vec<PublicKey> = Vec::new();
+
+			// Gathers all the pubkeys of the sync committee members that participated in siging the header.
+			for (bit, pubkey) in sync_committee_bits.iter().zip(sync_committee_pubkeys.iter()) {
+				if *bit == 1 as u8 {
+					let pubk = pubkey.clone();
+					participant_pubkeys.push(pubk);
+				}
+			}
+
+			let domain_type = DOMAIN_SYNC_COMMITTEE.to_vec();
+
+			// Domains are used for for seeds, for signatures, and for selecting aggregators.
+			let domain = Self::compute_domain(domain_type, Some(fork_version), validators_root)?;
+
+			let domain_bytes = domain.as_bytes();
+
+			// Hash tree root of SigningData - object root + domain
+			let signing_root = Self::compute_signing_root(header, domain)?;
+
+			let signing_root_bytes = signing_root.as_bytes();
+
+			// Verify sync committee aggregate signature.
+			Self::bls_fast_aggregate_verify(
+				participant_pubkeys,
+				signing_root,
+				sync_committee_signature,
+			)?;
+
+			Ok(())
+		}
+
+		pub(super) fn bls_fast_aggregate_verify(
+			pubkeys: Vec<PublicKey>,
+			message: H256,
+			signature: Vec<u8>,
+		) -> DispatchResult {
+			let sig = Signature::from_bytes(&signature[..]);
+
+			if let Err(e) = sig {
+				return Err(Error::<T>::InvalidSignature.into());
+			}
+
+			let agg_sig = AggregateSignature::from_signature(&sig.unwrap());
+
+			let public_keys_res: Result<Vec<milagro_bls::PublicKey>, _> =
+				pubkeys.iter().map(|bytes| milagro_bls::PublicKey::from_bytes_unchecked(&bytes.0)).collect();
+
+			if let Err(e) = public_keys_res {
+				match e {
+					AmclError::InvalidPoint => return Err(Error::<T>::InvalidSignaturePoint.into()),
+					_ => return Err(Error::<T>::InvalidSignature.into()),
+				};
+			}
+
+			let agg_pub_key_res = AggregatePublicKey::into_aggregate(&public_keys_res.unwrap());
+
```

Comment:
```suggestion
```

ditto: this empty line can be removed, since the error check directly relates to the preceding statement.